<a href="https://colab.research.google.com/github/joaovictor91/Linguagem_Promacao/blob/main/Jo%C3%A3o_Victor_Amaro_Franco_202_LP_Estrutura_N.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### 1. Elabore uma estrutura para representar um produto (código, nome, preço). Aplique 10% de aumento no preço do produto e apresente.

In [ ]:
class preco:
    codigo=0
    nome=''
    preco=0.0
    novo_preco=0.0

def main():
    produto=preco()
    produto.codigo=int(input("Informe o código do produto: "))
    produto.nome=input("Informe o nome do produto: ")
    produto.preco=float(input("Informe o nome do produto: R$"))
    produto.novo_preco=produto.preco+(produto.preco*10/100)
    print("O novo preço do",produto.nome,"é de R$",produto.novo_preco)

main()


### 2. Elabore uma estrutura para representar um produto (código, nome, preço). Cadastre 5 produtos, use vetor/lista. Aplique 10% de aumento no preço do produto e apresente todos os dados do preço.

In [ ]:
class preco:
    codigo=[]
    nome=[]
    preco_produto=[]
    novo_preco=[]

def main():
    produto=preco()
    total=5
    for i in range(total):
        
        produto.codigo.append(int(input("Informe o código do produto: ")))
        produto.nome.append(input("Informe o nome do produto: "))
        produto.preco_produto.append(float(input("Informe o nome do produto: R$")))
        r=produto.preco_produto[i]+(produto.preco_produto[i]*10/100)
        produto.novo_preco.append(r)

    for i in range(total):
        
        print(produto.codigo[i],"o produto",produto.nome[i],"é de R$",produto.novo_preco[i])

main()


Informe o código do produto: 1
Informe o nome do produto: 1
Informe o nome do produto: R$150
Informe o código do produto: 2
Informe o nome do produto: 2
Informe o nome do produto: R$9.50
1 o produto 1 é de R$ 165.0
2 o produto 2 é de R$ 10.45


### 3. Elabore uma estrutura para representar um produto (código, nome, preço). Crie uma função para cadastrar 5 produtos. Crie outra função para aplicar 10% de aumento no preço do produto e apresente todos os dados do preço.


In [ ]:
class preco:
    codigo=[]
    nome=[]
    preco_produto=[]
    novo_preco=[]

def aumento(preco_produto):
    r=preco_produto+(preco_produto*10/100)
    return (r)

def cadastro():

  produto=preco()
  total=2
  for i in range(total):

        
    produto.codigo.append(int(input("Informe o código do produto: ")))
    produto.nome.append(input("Informe o nome do produto: "))
    produto.preco_produto.append(float(input("Informe o nome do produto: R$")))
    
    produto.novo_preco.append(aumento(produto.preco_produto[i]))

  for i in range(total):
    print(produto.codigo[i],"o produto",produto.nome[i],"é de R$",produto.novo_preco[i])


def main():
  cadastro()
    

main()


Informe o código do produto: 1
Informe o nome do produto: 1
Informe o nome do produto: R$25
Informe o código do produto: 2
Informe o nome do produto: 2
Informe o nome do produto: R$99.99
1 o produto 1 é de R$ 27.5
2 o produto 2 é de R$ 109.98899999999999


### 4. Uma escola precisa montar o cadastro geral de seus alunos. Este cadastro deverá conter as seguintes informações por aluno: nome completo, data de nascimento, telefone, endereço completo e série atual. Levando em conta que esta escola possui no máximo 500 alunos, como você faria para estruturar estas informações num sistema de cadastro e consulta de cadastro, por nome e visualização de todos os dados? Implemente utilizando estrutura. Também use a criação de funções para cada operação. Use o menu a seguir:
Menu de opções:
1. Cadastrar alunos
2. Consulta por nome
3. Visualizar todos os dados
4. Sair


In [ ]:
class escola:
    nome=[]
    nascimento=[]
    endereco=[]
    serie=[]
    total_escola=499

def visualisar(aluno):
    print("Todas as aluns cadastradas")
    for i in range(len(aluno.nome)):
        print('Nome',aluno.nome[i],'\t Data de nascimento',aluno.nascimento[i],'\tEndereço do aluno',aluno.endereco[i],'\tSérie',aluno.serie[i])

    print("Operação realizada com sucesso")
    print(" ")
    
def busca(aluno):
    print("Pesquisa de Alunos")
    pesquisa=input("Digite um nome: ")
    for i in range (len(aluno.nome)):
        
        if pesquisa in aluno.nome[i]:

            n=aluno.nascimento[i]
            e=aluno.endereco[i]
            s=aluno.serie[i]

        
    if pesquisa in aluno.nome:
        print(pesquisa,"foi encontrado")
        print("Ele esutda na serie",s,"nasceu em",n,"seu endereço é",e)
            
    else:
        print(pesquisa,"não foi encontrado")

    print("Operação realizada com sucesso")
    print(" ")


             
def cadastrar():
    aluno=escola()
    total=25
    if len(aluno.nome)<=aluno.total_escola:

      print("Cadastro de alunos")
      for i in range(total):

        
        aluno.nome.append((input("Informe o nome do aluno: ")))
        aluno.nascimento.append(input("Informe a data de nascimento: "))
        aluno.endereco.append(input("Informe o endereço do aluno: "))
        aluno.serie.append(input("Informe a serie que o aluno está: "))
        
        
    else:
      print("Já foram realizados todos cadastrados")    
        
        
    
    print("Operação realizada com sucesso")
    print(" ")
    return(aluno)
    
        

def main():
    
       
        
    opcoes=0
    print("Menu da Escola")
    
    while opcoes!=4:
        
        
        print("Informe o que deseja fazer ")
        print("1-Cadastrar 15 Alunos")
        print("2-Visualizar todos os alunos")
        print("3-Consultar por nome")
        print("4-Sair")
        opcoes=(int(input("")))
        if  opcoes==1:
            conta=cadastrar()
            
            
        elif  opcoes==2:
            visualisar(conta)   
            
        elif opcoes==3:
            
            busca(conta)
            
        elif opcoes==4:

            
         print("Progama Finalizado")

        elif opcoes>4 or opcoes<=0:

          print("Opção Inválida")
          
    
            
main()


Menu da Escola
Informe o que deseja fazer 
1-Cadastrar 15 Alunos
2-Visualizar todos os alunos
3-Consultar por nome
4-Sair


KeyboardInterrupt: ignored

### 5. Faça um programa que realize o cadastro de contas bancarias com as seguintes informações: numero da conta, nome do titular e saldo. O banco permitirá o cadastramento de 15 contas. Crie uma função para cada opção do menu a seguir. Utilize a estrutura na passagem de parâmetro:
Menu de opções:
1. Cadastrar contas
2. Visualizar todas as contas
3. Consultar por nome
4. Alterar nome e/ou saldo de um número de conta
5. Excluir a conta com menor saldo
6. Sair



In [ ]:
class banco:
    nome=[]
    saldo=[]
    n=[]
    opcoes=0
    menor=0.0
    total=15

def visualisar(cliente):
    print("Todas as contas cadastradas")
    for i in range(len(cliente.nome)):
        print('Nº da Conta',cliente.n[i],'\tNome',cliente.nome[i],'\tsaldo R$',cliente.saldo[i])

    print("Operação realizada com sucesso")
    print(" ")
    
def busca(cliente):
    print("Pesquisa de clientes")
    pesquisa=input("Digite um nome: ")
    for i in range (len(cliente.nome)):
        
        if pesquisa in cliente.nome[i]:

            s=cliente.saldo[i]
            num=cliente.n[i]

        
    if pesquisa in cliente.nome:
        print(pesquisa,"foi encontrado")
        print("Nº da conta",num,"e o saldo é de R$",s)
            
    else:
        print(pesquisa,"não foi encontrado")

    print("Operação realizada com sucesso")
    print(" ")


def alterar(cliente):
    print("ALteração da conta do cliente")
    pesquisa=input("Digite um nome: ")
    for i in range (len(cliente.nome)):
        
        if pesquisa in cliente.nome[i]:
            alterar=input("Informe o novo nome: ")
            cliente.nome[i]=alterar
            print("Gostaria de alterar o saldo S/N")
            a=input("")
            if a=='s' or a=='S':
                modificar=float(input("Informe o saldo R$"))
                cliente.saldo[i]=modificar
            else:
                print("OK")
            

    if pesquisa not in cliente.nome[i]:
        print(pesquisa,"não foi encontrado")

    print("Operação realizada com sucesso")
    print(" ")

def retirar(cliente):
    
    print("Ciente com menor saldo foi excluido") 
    for i in range (len(cliente.nome)):
        
         
        if i==0:
             menor=cliente.saldo[i]

        if cliente.saldo[i]<menor:
             menor=cliente.saldo[i]

    for i in range (len(cliente.nome)):
        
         
        
        if menor == cliente.saldo[i]:
            
             
            pos=i

    cliente.nome.pop(pos)
    cliente.saldo.pop(pos)
    cliente.n.pop(pos)
    
        
    print("Operação realizada com sucesso")
    print(" ")
             
     
        
    
             
def cadastrar():
    cliente=banco()
    cont=0
    import random
    print("Cadastro de cliente")
    for i in range(cliente.total):
        
        cliente.nome.append(input("Informe o seu nome: "))
        cliente.saldo.append(float(input("Informe o seu saldo: R$")))
        
        cont=random.randrange(999, 9999)
        
        if cont not in cliente.n:
            
            cliente.n.append(cont)
        
        
        
    print("Operação realizada com sucesso")
    print(" ")
    return(cliente)
    
        

def main():
    
       
        
    opcoes=0
    print("Menu do Banco")
    
    while opcoes!=6:
        
        
        print("Informe o que deseja fazer ")
        print("1-Cadastrar 15 Contas")
        print("2-Visualizar todas as contas")
        print("3-Consultar por nome")
        print("4-Alterar nome e/ou saldo de um número de conta")
        print("5-Excluir a conta com menor saldo")
        print("6-Sair")
        opcoes=(int(input("")))
        if  opcoes==1:
            conta=cadastrar()
            
            
        elif  opcoes==2:
            visualisar(conta)   
            
        elif opcoes==3:
            
            busca(conta)
           
            opcoes=0
            
        elif opcoes==4:
            alterar(conta)
            
            opcoes=0

        elif opcoes==5:
            retirar(conta)
            
            opcoes=0
            
        if opcoes==6:
            
            print("Progama Finalizado")
          
    
            
main()


####6. Uma empresa prestadora de serviços armazena informações sobre os serviços prestados. Sabe-se que a empresa pode realizar no máximo três serviços diariamente. É de interesse de sua direção manter um histórico mensal (30 dias) sobre os serviços prestados.
A empresa realiza quatro tipos de serviços: 

1) pintura; 

2) jardinagem; 

3) faxina e

4) reforma em geral.

Cada serviço realizado deve ser cadastrado com as seguintes informações: número do serviço, valor do serviço, código do serviço e código do cliente.

Cadastre os quatro tipos de serviços (código e descrição) que a empresa poderá realizar. Para isso, utilize um vetor de quatro posições.
O programa deverá mostrar o seguinte menu de opções:
1. Cadastrar os tipos de serviços
2. Mostrar todos os tipos de serviço
3. Cadastrar os serviços prestados
4. Mostrar todos os serviços prestados
5. Mostrar os serviços prestados em determinado dia
6. Mostrar os serviços prestados dentro de um intervalo de valor
7. Mostrar um relatório geral (separado por dia), que exiba, inclusive, a descrição do tipo do serviço

**Para a opção 1**: deve-se cadastrar os tipos de serviços oferecidos pela empresa, com código e descrição.

**Para a opção 2**: deve-se considerar que deverão ser cadastrados os serviços prestados ao logo do mês. Em cada dia podem ser cadastrados, no máximo, três serviços prestados.

Utilize uma matriz capaz de armazenar em cada posição todas as informações referentes a um serviço prestado (número, valor, código do serviço, código do cliente). Cada linha representa um dia do mês. Dessa maneira, considere a matriz com dimensão 30 × 3.

Solicite o dia em que o serviço foi prestado e as demais informações.
Lembre-se de que a empresa só pode prestar os serviços que já tenham sido cadastrados no vetor de tipo de serviços.

Caso o usuário digite um código de tipo de serviço inválido, o programa deverá mostrar uma mensagem de erro.

Quando o usuário tentar cadastrar mais de três serviços prestados em um mesmo dia, também deverá mostrar uma mensagem de erro.

**Para a opção 3**: o programa deverá receber o dia que se deseja consultar e mostrar os respectivos serviços prestados.

**Para a opção 4**: o programa deverá receber o valor mínimo e o valor máximo e mostrar os serviços prestados que estiverem nesse intervalo.

**Para a opção 5**: o programa deverá mostrar todos os serviços prestados, conforme o exemplo a seguir.

                  DIA 01


No do serviço| Valor do serviço R$| Código do serviço| Descrição| Código do cliente
---|---|---|---|---
100| 200,00| 1| Pintura| 1
150| 100,00| 3| Faxina| 5

                  DIA 02

No do serviço| Valor do serviço R$| Código do serviço| Descrição| Código do cliente
---|---|---|---|---
301| 600,00| 4| Reforma em geral| 3
280| 352,00| 1| Pintura| 2

In [ ]:
class empresa:
  s=0
  cod=0
  nome_do_servico=''
  
  n_servico=0
  
class Cdastro_Servico:
    valor=0.0
    cod_cliente=0
    n_servico=0
    cod_servico=0
    
    
    

def cadastrar_servico(s):
  total_servico=4
  t=1
  cont=0 

class repetidos:
  rep=list()
class clientes_repetidos:
  
  repcod=list()
  
def cadastrar_servico(s):
  t=1
  total_servico=4
  cont=0
  print("Cadastro de serviço e digite o código do serviço dos números 1 ao 4")
  if len(s)<=t:

    for i in range(total_servico):
      e=empresa()
      cont=cont+1
      e.cod=cont
      
      e.nome_do_servico=input("Informe o nome do serviço: ")
      
      
      s.append(e)

  else:
    print("Todos os cadastros já foram realizados")

  return(s)
    
def visualisar(s):
  for i in range(len(s)):
    print('Nº de serviço',s[i].cod,'\tNome do serviço',s[i].nome_do_servico)
          

def validacao(s,codigo,servic,r0):
    
    while r0!=True:
        for i in range(len(s)):
            
            if codigo == s[i].cod:
                
                return (codigo)
                r0=True
                
                
        if codigo != s[i].cod:
          
          codigo=0
          r0=False
          print("Código Inválido")
          codigo=int(input("Informe o codigo do Serviço: "))
    


def valida_n(s,codigo):
  for i in range(len(s)):
    if codigo==s[i].cod:
      r=s[i].nome_do_servico
      return(r)
    
    
def servicos(s,cad):
  resp=' '
  total_dias=30
  t=3
  dia=1
  cont=0
  print("Cadastrados dos serviços ao longo do mês")
  di=int(input("Informe um o dia que gostaria de cadastrar: "))
  cont=0
  di=di-1
  r=repetidos()
  
  
  import random
  
  
      
      
  print("Cadastro de serviços")
  print("Serviços disponiveis")
  if di<30:
    
    
    
      
    for i in range(len(s)):
        print('Nº de serviço',s[i].cod,'\tNome do serviço',s[i].nome_do_servico)
    for i in range(total_dias):
      
      if di==i:
        if di not in r.rep:
          
          r.rep.append(di)
            
          print("Dia",dia)
          
          for j in range(t):
            rep=clientes_repetidos()
            servic=Cdastro_Servico()
            rpt=False
            cont+=1
            while rpt!=True:
              
              if cont not in rep.repcod:
                
                
                rep.repcod.append(cont)
                rpt=True
              elif cont in rep.repcod:

                cont+=1
                rpt=False
            servic.cod_cliente=cont
            cad[i][j].cod_cliente=servic.cod_cliente
            servic.valor=(float(input("Informe o valor do serviço: R$")))
            c=random.randrange(99, 999)
            
            cad[i][j].valor=servic.valor      
                  
            servic.n_servico=c
            
            cad[i][j].n_servico=servic.n_servico
            codigo=int(input("Informe o codigo do Serviço: "))
            r0=False
            r=validacao(s, codigo,servic,r0)
            
            servic.cod_servico=r
            print("servic.cod_servico",servic.cod_servico)
            cad[i][j].cod_servico=servic.cod_servico
            

        elif di in r.rep:
          print("Neste dias já foram realizados todos os cadastros")
            
      dia+=1    
      
      
         
      
  
      
        
  
  return(cad)
  
              

    
def visualisar_s(s,cad):
  total_dias=30
  t=3
  dia=1
  print("Todos os serviços cadastrados")
  for i in range(total_dias):
    
    for j in range(t):
      if cad[i][j].valor>0:
        print("Dia",dia)
        codigo=cad[i][j].cod_servico
        desc=valida_n(s,codigo)
                
        print('Cod do serviço','\t\t\tValor do serviço em R$','\t\t\tCod Serviço','\t\t\tDescrição','\t\t\tCódigo do Cliente')
        print(cad[i][j].n_servico,'\t\t\t\t\t',cad[i][j].valor,'\t\t\t\t\t',cad[i][j].cod_servico,'\t\t\t',desc,'\t\t\t\t',cad[i][j].cod_cliente)
      
      
    dia+=1

def visualisar_s_dia(s,cad):
  total_dias=30
  t=3
  dia=1
  print("Pesquisa por um determinado dia")
  d=int(input("Informe qual dia deseja buscar: "))
  d=d-1
  for i in range(total_dias):
    
    for j in range(t):
      
      if d==i:
        print("Dia",dia)
        codigo=cad[i][j].cod_servico 
        desc=valida_n(s,codigo)
                
        print('Cod do serviço','\t\t\tValor do serviço em R$','\t\t\tCod Serviço','\t\t\tDescrição','\t\t\tCódigo do Cliente')
        print(cad[i][j].n_servico,'\t\t\t\t\t',cad[i][j].valor,'\t\t\t\t\t',cad[i][j].cod_servico,'\t\t\t',desc,'\t\t\t\t',cad[i][j].cod_cliente)
    dia+=1  


                


def visualisar_s_valor(s,cad):
  total_dias=30
  t=3
  dia=1
  
  print("Pesquisa por um preço mínimo e um preço maximo")
  valor_min=float(input("Informe um valor minimo: "))
  valor_max=float(input("Informe um valor máximo: "))
  for i in range(total_dias):
                                
      for j in range(t):
          
          if valor_min<=cad[i][j].valor and valor_max>=cad[i][j].valor:
            print("Dia",dia)
              
            codigo=cad[i][j].cod_servico
                
            desc=valida_n(s,codigo)
                
            print('Cod do serviço','\t\t\tValor do serviço em R$','\t\t\tCod Serviço','\t\t\tDescrição','\t\t\tCódigo do Cliente')
            print(cad[i][j].n_servico,'\t\t\t\t\t',cad[i][j].valor,'\t\t\t\t\t',cad[i][j].cod_servico,'\t\t\t',desc,'\t\t\t\t',cad[i][j].cod_cliente)
                

      dia+=1


  
      
def visualisar_relátorio(s,cad):
  total_dias=30
  t=3
  dia=1
  d=0
  print("Relatório")
  
  for i in range(total_dias):
      print("Dia",dia)
      
                   
      for j in range(t):
          
          if d==i:
              
              if cad[i][j].valor>0:
                codigo=cad[i][j].cod_servico
                
                desc=valida_n(s,codigo)
                
                print('Cod do serviço','\t\t\tValor do serviço em R$','\t\t\tCod Serviço','\t\t\tDescrição','\t\t\tCódigo do Cliente')
                print(cad[i][j].n_servico,'\t\t\t\t\t',cad[i][j].valor,'\t\t\t\t\t',cad[i][j].cod_servico,'\t\t\t',desc,'\t\t\t\t',cad[i][j].cod_cliente)
                
                
         
      d+=1
      dia+=1

  
  

def main():
    total_dias=30
    t=3
    servico=[]   
    cadastrar=[]

    for i in range(total_dias):
          
          
          
        d=[]
          
                
        
        for j in range(t):
          servic=Cdastro_Servico()
          servic.valor=0.0
          servic.n_servico=0
          servic.cod_servico=0
          codigo=0
          r0=0
          r=0
          r2=0
          servic.cod_servico=r
          servic.nome_servico=r2
          d.append(servic)   
            
          
          
        cadastrar.append(d)
    opcoes=0
    print("Menu da Empresa")
    
    while opcoes!=8:
        
        
        print("Informe o que deseja fazer ")
        print("1-Cadastrar 4 serviços ")
        print("2-Mostrar todos tipos de serviços")
        print("3-Cadastrar os serviços prestados")
        print("4-Mostrar os serviços cadastrados")
        print("5-Mostrar os serviços prestados em determinado dia")
        print("6-Mostrar os serviços prestados dentro de um intervalo de valor")
        print("7-Mostrar um relatório geral (separado por dia), que exiba, inclusive, a descrição do tipo do serviço")
        print("8-Sair")
        opcoes=(int(input("")))
        if  opcoes==1:
            servico=cadastrar_servico(servico)
            
            
        elif  opcoes==2:
            visualisar(servico)
                        
        elif opcoes==3:
            
            cadastrar=servicos(servico,cadastrar)
            
            
            
            
            
        elif opcoes==4:
            visualisar_s(servico,cadastrar)
            
            

        elif opcoes==5:
            visualisar_s_dia(servico,cadastrar)
            

        elif opcoes==6:
                         
            visualisar_s_valor(servico,cadastrar)
            
        elif opcoes==7:
            
             
            visualisar_relátorio(servico,cadastrar)

             
        elif opcoes==8:
            
            print("Progama Finalizado")
            
        elif opcoes>8 or opcoes<=0:

          print("Opção Inválida")
          
            
       
          
    
            
main()


Menu da Empresa
Informe o que deseja fazer 
1-Cadastrar 4 serviços 
2-Mostrar todos tipos de serviços
3-Cadastrar os serviços prestados
4-Mostrar os serviços cadastrados
5-Mostrar os serviços prestados em determinado dia
6-Mostrar os serviços prestados dentro de um intervalo de valor
7-Mostrar um relatório geral (separado por dia), que exiba, inclusive, a descrição do tipo do serviço
8-Sair
1
Cadastro de serviço e digite o código do serviço dos números 1 ao 4
Informe o nome do serviço: Faxina
Informe o nome do serviço: Reforma
Informe o nome do serviço: Pintura
Informe o nome do serviço: Jardinagem
Informe o que deseja fazer 
1-Cadastrar 4 serviços 
2-Mostrar todos tipos de serviços
3-Cadastrar os serviços prestados
4-Mostrar os serviços cadastrados
5-Mostrar os serviços prestados em determinado dia
6-Mostrar os serviços prestados dentro de um intervalo de valor
7-Mostrar um relatório geral (separado por dia), que exiba, inclusive, a descrição do tipo do serviço
8-Sair
3
Cadastrados do